inc/admin-ui.php

```diff
@@ -287,3 +287,29 @@ function render_posts_column( $column, $post_id ): void {
 	}
 }
 add_action( 'manage_profile_posts_custom_column', __NAMESPACE__ . '\render_posts_column', 10, 2 );
+
+/**
+ * Remove the author support from post types.
+ * This is done to prevent the author dropdown from being displayed in the post editor.
+ */
+function remove_author_support() {
+	// Get all post types that support authors.
+	$post_types = get_post_types_by_support( 'author' );
+
+	// Add byline support to the list of supported post types.
+	add_filter( 'byline_manager_supported_post_types', fn() => $post_types );
+
+	/**
+	 * Filter the list of post types that should not have author support.
+	 * Defaults to posts that originally had author support.
+	 *
+	 * @param string[] $post_types Post types with author support removed.
+	 */
+	$post_types = apply_filters( 'byline_manager_remove_author_support', $post_types );
```

Comment:
Let's use the function in the Utils class that already does this instead of duplicating the logic here:
```suggestion
	/**
	 * Filter the list of post types that should not have author support.
	 * Defaults to all post types that support Byline Manager.
	 *
	 * @param string[] $post_types Post types with author support removed.
	 */
	$post_types = apply_filters( 'byline_manager_remove_author_support', Utils::get_supported_post_types() );
```

Reply:
@kevinfodness If we do that, then the byline manager functionality won't be added to the editor as we are removing byline manager support, so I think we need to remove author support while keeping the original post types with author support as supported by byline manager

Reply:
`get_supported_post_types` gets called later

Reply:
or apply `byline_manager_supported_post_types` from outside the plugin

Reply:
Ah, I see what you're saying. Let me think on it for a few and I'll get back to you.

Reply:
OK, what we need to do is to cache the lookup in the utils function so it doesn't zero itself out when we remove support. You should be able to change the `get_supported_post_types` function in the Utils class to the following and it should work:

```php
public static function get_supported_post_types(): array {
	static $post_types;
	if ( ! isset( $post_types ) ) {
		$post_types = get_post_types_by_support( 'author' );
	}

	/**
	 * Filter the list of supported post types. Defaults to all post types
	 * supporting 'author'.
	 *
	 * @param array $post_types Post types with which to use Byline Manager.
	 */
	return apply_filters( 'byline_manager_supported_post_types', $post_types );
}
```

Reply:
@kevinfodness, that's smart, thanks. Updated